### ISSUE TITLE
investigate whether we can provide specific feedback about CORS issues in the case of fetch cell errors

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#1396]
CORS issues are tricky and confusing. it would be great if we could let people know if they are hitting a CORS error in particular (rather than a fetch error in general) when they run into a CORS problem, and direct them to help / workarounds / more info.

(in investigating this, we might find that there are other kinds of errors we can surface/specify)

cc @hamilton

**Comment by bcolloran** [comment]
this link indicates that it may not be possible to detect a CORS problem from JS *at all* https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Functional_overview

we should poke a little more at this, but if it is the case, we should come up with some other UX treatment to help people

**Comment by bcolloran** [comment]
supercedes #1381

**Comment by hamilton** [comment]
I don't think there is genuinely any way for us to detect a CORS issue from the notebook itself given the sandboxing. As such, I might recommend we do something that React does – in the error message, provide a link to some additional documentation about what might have caused the `NetworkError when attempting to fetch resource` message, and they can click on the link and we can put the CORS issue at the very top of that bit of documentation, as well as some methods for detecting if there was a CORS issue (open up dev tools and investigate).

**Comment by hamilton** [comment]
(I see we already have a CORS Problems section in our docs, maybe just link to that)

**Comment by bcolloran** [comment]
yeah that might be about as good as we can do. sounds reasonable to me.

**Comment by bcolloran** [comment]
i think the best practical option is to add a message whenever there is a fetch error that we can't specify which says something like "failed to fetch resource at X. This may be a CORS issue, see the docs at Y."

**Comment by bcolloran** [comment]
@mdboom fyi about this issue -- if it's convenient and obvious how to add this link in the error message while you're looking at the fetch code, that would be great! but don't spend a lot of extra time on it if it's not clear

**Comment by yashdev9274** [comment]
is it still open, what is the issue after merge?
